FFMediaToolkit/Decoding/MediaFile.cs

```diff
@@ -17,38 +18,31 @@ private unsafe MediaFile(InputContainer container, MediaOptions options)
         {
             this.container = container;
 
-            if (container.Video != null)
+            var video = container.Decoders.Where(codec => codec.Info.Type == MediaType.Video);
+            var audio = container.Decoders.Where(codec => codec.Info.Type == MediaType.Audio);
+
+            if (video.Any())
             {
-                Video = new VideoStream(container.Video, options);
+                Video = video.Select(codec => new VideoStream(codec, options)).ToArray();
             }
 
-            if (container.Audio != null)
+            if (audio.Any())
             {
-                Audio = new AudioStream(container.Audio, options);
+                Audio = audio.Select(codec => new AudioStream(codec, options)).ToArray();
             }
 
             Info = new MediaInfo(container.Pointer);
         }
 
         /// <summary>
-        /// Gets the video stream.
+        /// Gets the video streams.
         /// </summary>
-        public VideoStream Video { get; private set; }
+        public VideoStream[] Video { get; }
 
         /// <summary>
-        /// Gets a value indicating whether the file contains video stream and the stream is loaded.
+        /// Gets the audio streams.
         /// </summary>
-        public bool HasVideo => Video != null;
```

Comment:
it's a public api that I'm not sure makes sense to remove, since it can be trivially implemented by checking if there are >0 video streams

Reply:
I debated about this with myself and removed it thinking the exact same thing, however I get where you're coming from, it doesn't make sense to make a breaking change unnecessarily.  I'll make sure to add it back in.